### DISCUSSION TITLE
Does ESP-NOW work with SPI protocol altogether?

### DISCUSSION THREAD

**Discussion by donskytech** [micropython/micropython#12994] [ESP32] [answered]
Hi,
I have set up my ESP-NOW between two ESP32 in my MicroPython Application.  I can retrieve my sensor readings and send them through ESP-NOW.

However, when I try to add a TFT LCD Display (ILI9314) to display my sensor readings, my application stops working and cannot work anymore because of the below problem.

```
rst:0xc (SW_CPU_RESET),boot:0x12 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:4728
load:0x40078000,len:14876
ho 0 tail 12 room 4
load:0x40080400,len:3368
entry 0x400805cc
```

Not sure why this is like this. Do you have any idea how to debug this type of problem?

Some of the troubleshooting that I did were:
1. Remove my TFT LCD Display code... (same error)
2. Remove the TFT LCD Display physically, I mean the wires and only the two ESP32 left..  (same error)

The issue persists.

The only way for me to make it work again is to erase and re- download the MicroPython firmware so that I could go back to a fresh start and then add my ESP-NOW code again.

**Comment by peterhinch** [comment, marked as answer]
We'll need more information to have any hope of diagnosing this:
* What type of ESP32 - specific board type.
* Type of sensor and driver in use.
* Pin allocations.
* What display driver are you using?
* Code.

If all the code is Python you should not need to re-flash the firmware. The best way to develop is to have the code started by a single module - let's say `my_code.py`. You can test it by issuing
```py
>>> import my_code
```
When it's debugged and you want it to run on power up, edit `main.py` to do the import.

That way you should never need to re-flash: a hard reset or power cycle should always get you back to the REPL.

**Reply by donskytech** [reply]
Hey, I found the issue while following your tips about having this single module entry point.  The issue is in my code about my file renaming.
Thanks for being helpful.  It seems to be working as expected now.
Kudos to you!
